### DISCUSSION TITLE
Security

### DISCUSSION THREAD

**Discussion by banannacodepie** [house-of-abbey/GarminHomeAssistant#117] [Q&A]
I'm trying this out and i'd like to see if I can get more info on securing my home assistant server while being able to use this application on my watch.
1. Is there anyway to secure the json file used to define the dashboard json? It seems to be publicly available on the server. Given we're accessing the server via an API token is there anyway tho keep this private?
2. I used a cloudflared tunnel to access my ha server. Clouldflare has some ways to limit who can access the tunnel by IP, geolocation or other authentication methods. Since my primary use for this tunnel is for garmin, would it be possible to limit the access to the tunnel to some garmin IP's? I'm not sure if garmin has rely servers or if the https requests come directly from the phone connected to the watch.
3. Is it possible to only export the restful API instead of the entire HA frontend? (this is probably more of a Homeassistant question instead of one specific to this app, but i figured I'd ask).

All in all this app seems great. Just would like to limit my ha exposure to the bare minimum required by this application.
I have other means to access my home assistant, so the exposure is soley to provide access to my garmin :)

**Comment by josephabbey** [comment]
To actually get any data or control anything, you must authenticate with homeassistant, so the only concern there is that someone could steal your watch. 

The dashboard JSON file is served by Home-Assistant as a static file, it does not require authentication and there is no simple way to do that, this is not a major security concern as it is useless without an access token, it is just a list of entities.

All in all, the only real security risk with the app is that your watch is not locked.

**Reply by banannacodepie** [reply]
Thanks for the feedback!
